contracts/roles/RevocationRegistryOnChain.sol

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity >= 0.7.0 < 0.9.0;
+
+import "@ensdomains/ens-contracts/contracts/registry/ENSRegistry.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/utils/introspection/ERC165Checker.sol";
+import "@ew-did-registry/proxyidentity/contracts/IOwned.sol";
+import "./RoleDefinitionResolver.sol";
+import {ClaimManager} from "./ClaimManager.sol";
+
+interface EthereumDIDRegistry {
+    function identityOwner(address identity) external view returns(address);
+    function validDelegate(address identity, bytes32 delegateType, address delegate) external view returns(bool);
+}
+    
+contract RevocationRegistryOnChain {
+    bytes32 constant private ASSERTION_DELEGATE_TYPE = 0x766572694b657900000000000000000000000000000000000000000000000000;
+
+    struct RevokedClaim {
+        address revoker;
+        uint revokedTimestamp;
+    }
+
+    mapping (bytes32 => mapping(address => RevokedClaim)) revokedClaims;
+    address private didRegistry;
+    address private ensRegistry;
+    address private claimManager;
+    
+    event Revoked(bytes32 role, address subject, address revoker);
+
+    constructor(address _didRegistry, address _ensRegistry, address _claimManager){
+        didRegistry = _didRegistry;
+        ensRegistry = _ensRegistry;
+        claimManager = _claimManager;
+    }
+
+    function revokeClaim(bytes32 role, address subject, address revoker) public {
+        require(revokedClaims[role][subject].revokedTimestamp == 0, "The claim is already revoked");
+        verifyRevoker(role, revoker);
+        revokedClaims[role][subject].revoker = revoker;
+        revokedClaims[role][subject].revokedTimestamp = block.number;
+        emit Revoked(role, subject, revoker);
+    }
+
+    function revokeClaimsInList(bytes32 role, address [] memory subjects, address revoker) public{
+        verifyRevoker(role, revoker);
+        for(uint i=0; i<subjects.length ; i++) {
+            require(revokedClaims[role][subjects[i]].revokedTimestamp == 0, "The claim is already revoked");
```

Comment:
Calling `revokeClaim(role, subjects[i], revoker)` for each iteration won't be an efficient way , also it is gas efficient to store the value there in the function rather than calling another function to do it. Please correct me if I am wrong.